test/framework/k8s_cluster.go

```diff
@@ -1248,13 +1249,38 @@ func (c *K8sCluster) K8sVersionCompare(otherVersion string, baseMessage string)
 	return version.Compare(semver.MustParse(otherVersion)), fmt.Sprintf("%s with k8s version %s", baseMessage, version)
 }
 
-func printDeploymentConditions(c *K8sCluster, namespace string, name string) {
+func printDetailedPodInfo(testingT testing.TestingT, kubectlOptions *k8s.KubectlOptions, podError error, pod v1.Pod) error {
```

Comment:
Hey a while back I looked at adding this sort of thing inside terratest: https://github.com/gruntwork-io/terratest/pull/1252 I never got time to progress on this PR. Let me know if you want to take it over :)

Reply:
Wow, exactly the same thing!

Reply:
Yeah, I'll definitely take a look at that PR and see how can I help